torch/csrc/autograd/profiler.h

```diff
@@ -203,23 +203,89 @@ TORCH_API void mark(std::string name, bool include_cuda = true);
 TORCH_API void pushRange(std::string name);
 TORCH_API void popRange();
 
-struct TORCH_API RecordFunction {
-  explicit RecordFunction(Function* fn);
+using TensorSizesList = std::vector<at::IntArrayRef>;
 
-  explicit RecordFunction(std::string name);
+struct TORCH_API FunctionCallContext {
+  explicit FunctionCallContext(Function* fn);
+  explicit FunctionCallContext(std::string name, int64_t sequence_nr = -1);
+  explicit FunctionCallContext(const char* name_ptr, int64_t sequence_nr = -1);
 
-  explicit RecordFunction(const char* name);
+  Function* func() const {
+    return fn_;
+  }
 
-  explicit RecordFunction(const char* name, int64_t current_sequence_nr);
+  const char* name() const {
+    return name_ptr_;
+  }
+
+  int64_t seqNr() const {
+    return sequence_nr_;
+  }
 
-  ~RecordFunction() {
-    popRange();
+  bool hasOwnedName() const {
+    return owned_name_ != nullptr;
   }
+
+  void setInputSizes(const TensorSizesList& sizes) {
+    input_sizes = c10::guts::make_unique<TensorSizesList>();
+    *input_sizes = sizes;
+  }
+
+  const TensorSizesList* inputSizes() const {
+    return input_sizes.get();
+  }
+
+ private:
+  Function* fn_ = nullptr;
+  std::unique_ptr<std::string> owned_name_;
+  const char* name_ptr_ = nullptr;
+  int64_t sequence_nr_ = -1;
+  std::unique_ptr<TensorSizesList> input_sizes;
```

Comment:
If we want to implement tracing on top of this we'll need exact inputs (probably packed as `IValue`s), not only their sizes.

Reply:
Yes, I agree, will remove the generalized UpdateFunctionCallContext and replace it with something more specific, it's main use case is inputs. Sometimes we have inputs ready as a stack of IValues (in JIT interpreter) but in VariableType we just have a codegen'd list of function arguments.